About passing parameters to sourced scripts (#6516)
MicrosoftDocs/PowerShell-Docs: reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md

#### Comma operator `,`

As a binary operator, the comma creates an array or appends to the array being

> created. In expression mode, as a unary operator, the comma creates an array

Reviewer: ```suggestion
created. In expression mode, or as a unary operator, the comma creates an array
```

Author: The point is, the unary comma is a parse error in argument mode.

Reviewer: I do see that and I was trying (poorly) to ensure that the unary operator portion still existed outside of  the expression mode as both `(,1)` in expression mode and `,1` outside expression mode use it as a unary operator. The way your change reads only references it inside expression mode siloing it.

> As a binary operator, the comma creates an array or appends to the array being
> created. In expression mode, as a unary operator, the comma creates an array
> with just one member. Place the comma before the member.

Reviewer 2: ```suggestion
As a binary operator in an expression, the comma creates an array or appends to the array being created. As a unary operator, placed before a member, the expression creates an array with just one member. Since `Write-Object` expects an argument, you must put the expression in parentheses.
```

```powershell
$myArray = 1,2,3
$SingleArray = ,1

> Write-Output (,1)

Reviewer: ```suggestion
$ExpressionSingleArray =  (,1)
```

Author: The point is, `Write-Output ,1` does not work, so you have to force expression mode here.

```

#### Dot sourcing operator `.`

<!--
01234567890123456789012345678901234567890123456789012345678901234567890123456789
-->
Runs a script in the current scope so that any functions, aliases, and
variables that the script creates are added to the current scope, overriding
existing ones. Parameters declared by the sourced script become variables too.
Parameters for which no value has been given become variables with no value.
However, the automatic variable `$args` is preserved.

```powershell

> . c:\scripts\sample.ps1 1 2 -Also:3

Reviewer: ```suggestion
. c:\scripts\sample.ps1 1 2 -Also 3
```

Author: Why exactly?  `-Also 3` is ambiguous, **Also** might be a switch.

Reviewer: I think this example is unnecessarily complex. `1` and `2` could be positional params  but without an actuall script to match to, the user won't know and really not have a reference. They aren't needed to highlight the behavior you are talking about. I think having something staight forward is a better fit here where it's extremely shallow in informational depth. 

`. c:\scripts\sample.ps1 -TestParam 'TestValue'`

It could be read as a switch but in reality, that's not the standard and there is no specific reference to a switch you are trying to highlight. We should be trying to highlight that the parameter is given a value and the user can understand that `-TestParam` will be available as `$TestParam`.  I don't see how `1` or `2` help highlight the behavior above?

Reviewer 2: I think we need to show the contents of a simple script and the results of dot-sourcing.

sample.ps1
```
[cmdletbinding()]
param([int]$timeout)

"The timeout value is $timeout."
```

Execution example:
```
PS> .\sample.ps1 -timeout 4
The timeout value is 4.
PS> $timeout
PS> . .\sample.ps1 -timeout 4
PS> $timeout
4
```

> . c:\scripts\sample.ps1 1 2 -Also:3

Reviewer 2: ```suggestion
. c:\scripts\sample.ps1 1 2 -TimeoutInSeconds 5
```
